app.territoiresentransitions.react/src/app/pages/collectivite/PlansActions/ToutesLesFichesAction/FichesActionListe.tsx

```diff
@@ -129,44 +144,100 @@ const FichesActionListe = ({
     setCurrentPage(1);
   }, [debouncedSearch]);
 
+  useEffect(() => {
+    if (!isGroupedActionsOn) setSelectedIds([]);
+  }, [isGroupedActionsOn]);
+
   const countTotal = data?.count || 0;
 
   return (
     <>
-      <div className="flex items-center gap-8 py-6 border-y border-primary-3">
-        {/** Tri */}
-        <div className="w-64">
-          <Select
-            options={sortOptions}
-            onChange={(value) =>
-              value && setSort(sortByOptions.find((o) => o.field === value)!)
-            }
-            values={sort.field}
-            customItem={(v) => <span className="text-grey-8">{v.label}</span>}
-            disabled={sortOptions.length === 1}
-            small
-          />
+      <div className="relative">
+        <div className="relative z-[1] bg-grey-2 flex max-xl:flex-col justify-between xl:items-center gap-4 py-6 border-y border-primary-3">
```

Comment:
Cool le responsive de cette partie ;)

Juste pourquoi avoir mis un `z-index` ici ? Quand je l'enlève ça change rien, je rate peut-être quelque chose.

Il faut au maximum éviter de mettre des z-index c'est hyper galère à debuguer.

Reply:
J'ai ajouté un sous menu "x actions sélectionnées / y actions" qui se déplie lorsqu'on clique sur le toggle. Sans le z-index, ce menu apparait superposé au menu des filtres lorsqu'il est replié.

Reply:
C'est pas très propre de cacher des éléments avec des `z-index` et c'est assez galère à débuguer.
Ce problème de faire des transitions couplées d'un `display: none` ou `position: absolute` ou `visibility: hidden` va se poser à chaque fois que l'on veut faire ce genre de transition.
Je suis assez d'acc que cet exemple est simple mais ça pourrait arriver sur des composants plus lourds où l'on ne veut pas charger le contenu inutilement. Je ne pense pas qu'avoir une stratégie plus solide pour ces cas soit overkill, surtout si la solution retenue est d'utiliser les `z-index`.

Après investigations je vois qu'il y a quelque chose de prévu en ce sens dans la [prochaine version de tailwind](https://tailwindcss.com/docs/v4-beta#starting-style-variant) encore en beta, dommage.

Sinon on peut faire un ptit wrapper pour appliquer les transitions que l'on veut, j'ai demandé à l'ia de faire ce que j'avais en tête, je pense qu'on peut améliorer cela pour avoir les transitions que l'on veut mais je trouve que c'est pas mal et que ça permet d'avoir l'animation tout en ayant l'enfant enlevé du dom lorsque non visible. Plus tard on pourrait remplacer ce composant par la feature tailwind de la v4 facilement plutôt qu'aller chercher des `z-index`.

```
const AnimatedDisplay = ({ isVisible, children, duration = 300 }) => {
  const [shouldRender, setShouldRender] = useState(isVisible);

  useEffect(() => {
    if (isVisible) {
      setShouldRender(true);
    } else {
      const timer = setTimeout(() => setShouldRender(false), duration);
      return () => clearTimeout(timer);
    }
  }, [isVisible, duration]);

  const classes = `
    transition-opacity ease-in-out
    ${isVisible ? 'opacity-100' : 'opacity-0'}
  `;

  return (
    shouldRender && (
      <div
        className={classes}
        style={{ transitionDuration: `${duration}ms` }}
      >
        {children}
      </div>
    )
  );
};

export default AnimatedDisplay;
```

Ou sinon on se dit que les micro-animations c'est pas ça qui rend le produit meilleur, c'est sympa mais c'est nice-to-have. Du moins le temps que la feature de la v4 de tailwind sorte.

Qu'en dites vous @marc-rutkowski, @elisfainstein, @farnoux ?

Reply:
Si on veut commencer à mettre de la micro ux (transitions), c'est un sujet qui va revenir.
D'ailleurs c'est pas la première fois que l'on rencontre ce cas de figure, voir le [SidePanel](https://github.com/incubateur-ademe/territoires-en-transitions/blob/fiches-pdf-groupe/app.territoiresentransitions.react/src/app/pages/collectivite/PlansActions/FicheAction/SideMenu.tsx#L53) qui permet d'associer des indicateurs à une fiche action. Tu avais déjà utilisé une logique similaire à ce que je propose, ça permettrait d'avoir une seule logique d'animation à l'affichage/masquage d'un élément en attendant la v4 de tailwind et sans utilisé de `z-index`.
Encore une fois, le `z-index` est très pénible à débuguer et est principalement utilisé par les éléments floating-ui actuellement.
Sinon on peut se noter de le faire plus tard mais je trouve ça dommage.

Reply:
Je suis d'accord avec Arthur, pour ce qui est d'éviter d'utiliser le `z-index` pour ce cas d'usage.
Et avec Marine pour dire que ce n'est pas prioritaire.
Mais si on veut vraiment mettre un peu d'animation (avec modération !) sur nos composants j'opterai pour l'utilisation d'une lib spécialisée comme [motion](https://motion.dev/docs/react-quick-start#exit-animations) plutôt que d'essayer de bricoler notre propre solution.

Reply:
Pour avoir fait pas mal d'animation dans le passé, c'est vraiment pas très compliqué surtout pour ces cas d'usages. Je ne suis pas pour mettre une nouvelle lib pour ces animations très simples. De plus la v4 de tailwind va nous permettre de faire exactement cela, la version beta est sortie, la release officielle ne devrait pas tarder.
Si ce n'est pas demandé, je serais d'avis de dropper les animations en attendant la v4 si on décide de ne pas faire de wrapper.

Reply:
Plutôt d'avis de merger cette PR en l'état parce qu'effectivement bloquée depuis un moment (et ça bloque aussi la sortie des autres sujets actions groupées), tout en gardant en tête que le sujet reste ouvert. Le mieux serait peut être de se prendre un petit temps d'équipe de vive voix pour s'aligner sur l'approche à avoir. @cparthur @farnoux